src/libraries/System.Private.CoreLib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -51,8 +51,14 @@ public Dictionary(int capacity, IEqualityComparer<TKey>? comparer)
 
             if (capacity > 0)
             {
-                Initialize(capacity);
+                Initialize(HashHelpers.GetPrime(capacity));
             }
+            else
+            {
+                InitializeEmpty();
```

Comment:
This is replaced with eventual calls to `Initialize(0)`. Note however that `Initialize(0)` will allocate arrays of size `3` instead of `1` in this case. What is the motivation behind this change?

Reply:
The idea is that using `InitializeEmpty` will reduce construction costs since it's just initializing with statically initialized arrays. If we called `Initialize(0)` we'd have to pay those allocation costs upfront even if the dictionary is never written to. It also allows us to maintain current binary serialization logic of not serializing entries if the dictionary hasn't been written to and constructed with a capacity of 0, either explicitly or implicitly.